registry/storage/driver/storagedriver.go

```diff
@@ -34,7 +34,14 @@ func (version Version) Minor() uint {
 const CurrentVersion Version = "0.1"
 
 // StorageDriver defines methods that a Storage Driver must implement for a
-// filesystem-like key/value object storage.
+// filesystem-like key/value object storage. Storage Drivers are automatically
+// registered via an internal registration mechanism, and generally created
+// via the StorageDriverFactory interface (https://godoc.org/github.com/docker/distribution/registry/storage/driver/factory).
+// See below for an example of how to get a StorageDriver for S3:
+//
+//	import _ "github.com/docker/distribution/registry/storage/driver/s3-aws"
```

Comment:
I don't think this example belongs at the top-level package. The use of factories are not required to use storage drivers.

Maybe just a pointer here to the factory package as a hint. This way, we avoid having to maintain this example in two places.